### ISSUE TITLE
a1aa0ce96 breaks t/run/switchF2.t on threaded builds under miniperl

### ISSUE THREAD

**Issue by jkeenan** [Perl/perl5#20524]
`t/run/switchF2.t` is now experiencing test failures when run against a **threaded build** of `miniperl`.  (It passes under `perl`.) The failures look like this:
```
$ ./miniperl -Ilib t/run/switchF2.t
ok 1 - passing -F implies -a
ok 2 - passing -F implies -an
ok 3 - passing -a implies -n
not ok 4 - PL_splitstr freed in each thread
# Failed test 4 - PL_splitstr freed in each thread at ./test.pl line 1177
#      got "Can\'t load module threads, dynamic loading not available in this perl.\n  (You may need to build a new perl executable which either supports\n  dynamic loading or has the threads module statically linked into it.)\n at - line 0.\nCompilation failed in require.\nBEGIN failed--compilation aborted."
# expected "b"
# PROG: 
# print $F[1]; threads->new(sub {})->join
# STATUS: 65280
ok 5 - PL_splitstr freed on extra -F switch
1..5
```
The relevant section of `t/run/switchF2.t`:
```
 29 SKIP:
 30 {
 31   $have_config or skip "Can't check if we have threads", 1;
 32   $Config::Config{usethreads} or skip "No threads", 1;
 33   # this would only fail under valgrind/ASAN
 34   fresh_perl_is('print $F[1]; threads->new(sub {})->join', "b",
 35                 {
 36                     switches => [ "-F,", "-Mthreads" ],
 37                     stdin => "a,b,c",
 38                 }, "PL_splitstr freed in each thread");
 39 }
```
Manual bisection points to recent commit a1aa0ce969:
```
commit a1aa0ce969357a471f335bc9f9362f49f178b1fe
Author:     Tony Cook <tony@develop-help.com>
AuthorDate: Mon Sep 26 15:03:50 2022 +1000
Commit:     Tony Cook <tony@develop-help.com>
CommitDate: Wed Nov 2 09:35:07 2022 +1100

    We freed PL_splitstr in each thread, allocate it in each thread too
```
@tonycoz, can you take a look?

**Comment by Leont** [comment]
That test should be skipped `if is_miniperl()`

**Comment by jkeenan** [comment]
> That test should be skipped `if is_miniperl()`

Thanks.  Please see https://github.com/Perl/perl5/pull/20525.

**Comment by jkeenan** [comment]
Suggestion from @Leont merged in decc3ce.  Closing ticket.
